mcs/class/System/System.Net/MacProxy.cs

```diff
@@ -688,6 +688,24 @@ static CFProxyType CFProxyTypeToEnum (IntPtr type)
 			if (type == kCFProxyTypeSOCKS)
 				return CFProxyType.SOCKS;
 			
+			//in OSX 10.13 pointer comparison didn't work for kCFProxyTypeAutoConfigurationURL
+			var typeString = CFString.AsString(type);
```

Comment:
Is this a cast or an allocate? Do we need to clean it up?

Reply:
I'll change it use CFStringCompare

https://developer.apple.com/documentation/corefoundation/1542911-cfstringcompare?language=objc